pkg/cluster/components/daemonset.go

```diff
@@ -0,0 +1,238 @@
+package components
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util/deployment"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+// DaemonSetUpdater can provide rolling updates on
+// the given DaemonSet.
+type DaemonSetUpdater struct {
+	// name of the DaemonSet object.
+	name string
+	// client is an API Server client.
+	client clientset.Interface
+	// pods is backed by an informer, contains list of Pods
+	// that belong to the DaemonSet.
+	pods StoreToPodLister
+	// daemonsets is a cache of DaemonSets backed by an informer.
+	daemonsets cache.StoreToDaemonSetLister
+	// selector is the DaemonSet selector.
+	selector labels.Selector
```

Comment:
`unversioned.LabelSelector` is a different type from `labels.Selector` which is what we need. Keeps us from having to convert every time.